dwave/plugins/torch/samplers/block_spin_update.py

```diff
@@ -0,0 +1,165 @@
+from collections import defaultdict
+from time import perf_counter
+
+import dwave_networkx as dnx
+import torch
+from networkx import Graph
+from torch import nn
+
+from dwave.plugins.torch.models.boltzmann_machine import GraphRestrictedBoltzmannMachine as GRBM
+from dwave.plugins.torch.utils import bit2spin_soft, rands
+
+
+class BlockSpinUpdateSampler(nn.Module):
+    """A block-spin update sampler for Chimera-, Pegasus-, and Zephyr-structured graph-restricted Boltzmann machines.
+
+    Note this is a sampler and not a neural network. It extends `nn.Module` for the convenience of
+    managing devices of indices (stored as parameters).
+
+    Due to the sparse definition of GRBMs (for better or worse), some tedious, and ugly, indexing
+    tricks were employed. Ideally, an adjacency list can be used, however, adjacencies are ragged,
+    which makes vectorization inapplicable.
+
+    Block-Gibbs and Block-Metropolis obey detailed balance and are ergodic methods at finite
+    temperature, which at fixed parameters converge upon Boltzmann distributions. Block-Metropolis
+    allows higher acceptance rates for proposals (faster single-step mixing), but is non-ergodic in
+    the limit of zero temperature. Decorrelation from an initial condition can be slower.
+    Block-Gibbs represents best practice for independent sampling.
+
+    Args:
+        G (Graph): A Chimera, Pegasus, or Zephyr graph.
+        grbm (GRBM): The Graph-Restricted Boltzmann Machine to sample from.
+    """
+
+    def __init__(self, G: Graph, grbm: GRBM, num_reads: int, kind: str):
```

Comment:
I think 'kind' is called 'proposal_acceptance_criteria' in SimulatedAnnealingSampler. It took me a while to work out what kind was, and would be good to standardize.

Reply:
Add 'Gibbs' as default